app-dashboard/js/view/map.js

```diff
@@ -46,8 +48,37 @@ define([
             this.map.on('draw:created', (e) => {
                 that.drawGroup.clearLayers();
                 that.drawGroup.addLayer(e.layer);
-                dispatcher.trigger('map:update-polygon', that.postgrestFilter());
+                var popupContent = '<form role="form" id="draw-form" enctype="multipart/form-data" class="form-horizontal">' +
+                    '<div class="form-group">' +
+                    '<input type="checkbox" id="enable_forecast_date" onchange="$(\'#forecast_date\').prop(\'disabled\', function(i, v) { return !v; })">&nbsp;<label for="forecast_date">Forecast date: </label>' +
+                    '<input class="form-control" type="text" id="forecast_date" disabled><br/>' +
+                    '<input type="checkbox" id="enable_station" onchange="$(\'#station\').prop(\'disabled\', function(i, v) { return !v; })">&nbsp;<label for="station">Station: </label><input class="form-control" type="text" id="station" disabled><br/>' +
+                    '<button type="submit" value="submit" class="btn btn-primary">Save</button>' +
+                    '<button type="button" id="cancel-draw" class="btn btn-default">Cancel</button>' +
+                    '</div></form>';
+                that.drawGroup.bindPopup(popupContent,{
+                    keepInView: true,
+                    closeButton: false,
+                    closeOnClick: false
+                    }).openPopup();
+
+                $('#forecast_date').datepicker({
+                    language: 'en',
+                    timepicker: true
+                });
+
+                $("#draw-form").submit(function(e){
+                    e.preventDefault();
+                    dispatcher.trigger('map:update-polygon', that.postgrestFilter());
```

Comment:
could you move the form creation into the backbone view?
example:
polygonForm = new FloodPolygonForm()

let additional data = polygonForm.postgrestFilter()
and in here, add like "dispatcher.trigger('map:update-polygon', that.postgrestFilter(), additional data);"